json/json.mbti

```diff
@@ -30,12 +41,7 @@ impl JsonValue {
   value(Self, String) -> Self?
 }
 
-pub enum ParseError {
-  InvalidChar(Position, Char)
-  InvalidEof
-  InvalidNumber(Position, String)
-  InvalidIdentEscape(Position)
-}
+pub type! ParseError JsonParseError
 impl ParseError {
```

Comment:
1.
todo:
type! ParseError (~a : Int, ~b : Int)
type! ParseError (Int, Int)
2. 
type! ParseError ErrorData
3.
impl Show for Error